specification/maintenance/Maintenance.Management/ApplyUpdate.tsp

```diff
@@ -0,0 +1,144 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.Maintenance;
+/**
+ * Apply Update request
+ */
+@subscriptionResource
+model ApplyUpdate
+  is Azure.ResourceManager.ProxyResource<ApplyUpdateProperties> {
+  ...ResourceNameParameter<
+    Resource = ApplyUpdate,
+    KeyName = "applyUpdateName",
+    SegmentName = "applyUpdates",
+    NamePattern = ""
+  >;
+}
+
+@armResourceOperations
+interface ApplyUpdateOps
+  extends Azure.ResourceManager.Legacy.CustomLegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+
+        /**
+         * Resource provider name
+         */
+        @path @segment("providers") providerName: string,
+
+        /**
+         * Resource parent type
+         */
+        @path resourceParentType: string,
+
+        /**
+         * Resource parent name
+         */
+        @path resourceParentName: string,
+
+        /**
+         * Resource type
+         */
+        @path resourceType: string,
+
+        /**
+         * Resource name
+         */
+        @path resourceName: string,
+
+        ...Azure.ResourceManager.Legacy.Provider,
+      },
+      KeysOf<ResourceNameParameter<
+        Resource = ApplyUpdate,
+        KeyName = "applyUpdateName",
+        SegmentName = "applyUpdates",
+        NamePattern = ""
+      >>,
+      MaintenanceError
+    > {}
+
+@armResourceOperations
+interface ApplyUpdates {
+  /**
+   * Track maintenance updates to resource with parent
+   */
+  getParent is ApplyUpdateOps.Read<ApplyUpdate>;
+}
+@armResourceOperations
+interface ApplyUpdateOperationGroupOps
+  extends Azure.ResourceManager.Legacy.CustomLegacyOperations<
+      {
+        ...ApiVersionParameter,
+        ...SubscriptionIdParameter,
+        ...ResourceGroupParameter,
+
+        /**
+         * Resource provider name
+         */
+        @path @segment("providers") providerName: string,
+
+        /**
+         * Resource parent type
+         */
+        @path resourceType: string,
+
+        /**
+         * Resource parent name
+         */
+        @path resourceName: string,
+
+        ...Azure.ResourceManager.Legacy.Provider,
+      },
+      KeysOf<ResourceNameParameter<
+        Resource = ApplyUpdate,
+        KeyName = "applyUpdateName",
+        SegmentName = "applyUpdates",
+        NamePattern = ""
+      >>,
+      MaintenanceError
+    > {}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/no-resource-delete-operation"
+@armResourceOperations
+interface ApplyUpdateOperationGroup {
+  /**
+   * Track maintenance updates to resource
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @operationId("ApplyUpdates_Get")
```

Comment:
We should consider using the new `@moveTo` decorator in TCGC when this is released.